pkg/detectors/doppler/doppler.go

```diff
@@ -44,20 +55,44 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		s1 := detectors.Result{
 			DetectorType: detectorspb.DetectorType_Doppler,
 			Raw:          []byte(resMatch),
+			ExtraData:    map[string]string{},
 		}
 
 		if verify {
-			req, err := http.NewRequestWithContext(ctx, "GET", "https://api.doppler.com/v3/workplace", nil)
+			req, err := http.NewRequestWithContext(ctx, "GET", "https://api.doppler.com/v3/me", nil)
 			if err != nil {
 				continue
 			}
 			req.Header.Add("Accept", "application/json")
-			req.SetBasicAuth(resMatch, "")
+			req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", resMatch))
 			res, err := client.Do(req)
 			if err == nil {
 				defer res.Body.Close()
 				if res.StatusCode >= 200 && res.StatusCode < 300 {
 					s1.Verified = true
+
+					body, err := io.ReadAll(res.Body)
+					if err != nil {
+						body = []byte{}
+					}
+
+					var data map[string]interface{}
```

Comment:
**suggestion**: The API spec indicates that `type` will be a `string`. We could `var data map[string]string` instead of `interface`. This way we can avoid casting to strings later in the code.

<img width="663" alt="Screenshot 2024-01-23 at 8 51 21 AM" src="https://github.com/trufflesecurity/trufflehog/assets/15034943/4a7b1c04-1e23-420f-9d48-a4f13f586aa6">

Alternatively, and thinking about this a bit more, we could just shove the entire body into a response struct (only defining the data we care about) rather than relying on multiple maps. https://mholt.github.io/json-to-go/ is a useful tool for converting json to go.

Reply:
https://github.com/trufflesecurity/trufflehog/pull/1945/files is a good example on how to load the response into a struct

Reply:
updates added.